src/asm.rs

```diff
@@ -1,179 +1,66 @@
 //! Miscellaneous assembly instructions
 
+// When inline assembly is enabled, pull in the assembly routines here. `call_asm!` will invoke
+// these routines.
+#[cfg(feature = "inline-asm")]
+#[path = "../asm/inline.rs"]
+pub(crate) mod inline;
+
 /// Puts the processor in Debug state. Debuggers can pick this up as a "breakpoint".
 ///
 /// **NOTE** calling `bkpt` when the processor is not connected to a debugger will cause an
 /// exception.
 #[inline(always)]
 pub fn bkpt() {
-    match () {
-        #[cfg(all(cortex_m, feature = "inline-asm"))]
-        () => unsafe { llvm_asm!("bkpt" :::: "volatile") },
-
-        #[cfg(all(cortex_m, not(feature = "inline-asm")))]
-        () => unsafe {
-            extern "C" {
-                fn __bkpt();
-            }
-
-            __bkpt();
-        },
-
-        #[cfg(not(cortex_m))]
-        () => unimplemented!(),
-    }
```

Comment:
I recently dug this up for unrelated reasons.  I just wanted to mention, this match construct was introduced because of rust-embedded/cortex-m#63 with the actual change being from rust-embedded/cortex-m#71.  Is it no longer relevant?

Reply:
It still builds fine for me. Maybe not with `inline-asm`. I should be able to add this feature back though.

Reply:
Ah, I remember now, I removed this to avoid all the warnings this generates. These were the reason almost the entire crate used function arguments like `_val`. I think we should leave this as-is for now.